tests/rados/test_osd_full.py

```diff
@@ -694,3 +674,157 @@ def check_full_warning(type, a_set):
+            cmds = [
+                "ceph osd set-nearfull-ratio 0.65",
+                "ceph osd set-backfillfull-ratio 0.69",
+                "ceph osd set-full-ratio 0.70",
+            ]
+            [cephadm.shell(args=[cmd]) for cmd in cmds]
+
+            # determine the number of objects to be written to the pool
+            # to achieve nearfull, backfillfull and full state.
+            full_objs = int(primary_osd_size * 0.7 / bench_obj_size_kb) + 1
+
+            # reweight osds to 1
+            [
+                cephadm.shell([f"ceph osd reweight osd.{osd_id} 1"])
+                for osd_id in acting_set1
+            ]
+
+            # perform rados bench to trigger full warning
+            full_config = {
+                "seconds": 200,
+                "b": f"{bench_obj_size_kb}KB",
+                "no-cleanup": True,
+                "max-objects": full_objs,
+            }
+            bench_obj.write(client=client_node, pool_name=pool1_name, **full_config)
+
+            log.info(f"{full_objs + 1} objects have been written to pool {pool1_name}")
+
+            check_full_warning(type="full", a_set=acting_set1)
+
+            # perform rados bench to trigger full warning
+            pool2_config = {
+                "seconds": 60,
+                "b": f"{bench_obj_size_kb}KB",
+                "no-cleanup": True,
+            }
+            bench_obj.write(client=client_node, pool_name=pool2_name, **pool2_config)
+            bench_obj.write(client=client_node, pool_name=pool3_name, **pool2_config)
+
+            time.sleep(7)  # blind sleep to let all objs show up in ceph df stats
+            pool_stat = rados_obj.get_cephdf_stats(pool_name=pool2_name)
+            if pool_stat["stats"]["objects"] == 0:
+                log.error(f"Pool {pool2_name} is not having any objects")
+                raise Exception(
+                    f"{pool2_name} expected to have some objects,"
+                    f" had {pool_stat['stats']['objects']} objects"
+                )
+            log.info(
+                f"{pool_stat['stats']['objects']} objects have been written to pool {pool2_name}"
+            )
+
+            pool_stat = rados_obj.get_cephdf_stats(pool_name=pool3_name)
+            if pool_stat["stats"]["objects"] == 0:
+                log.error(f"Pool {pool3_name} is not having any objects")
+                raise Exception(
+                    f"{pool3_name} expected to have some objects,"
+                    f" had {pool_stat['stats']['objects']} objects"
+                )
+            log.info(
+                f"{pool_stat['stats']['objects']} objects have been written to pool {pool3_name}"
+            )
+        except Exception as e:
+            log.error(f"Failed with exception: {e.__doc__}")
+            log.exception(e)
+            return 1
+        finally:
+            # deleting the created pool
+            rados_obj.detete_pool(pool=pool1_name)
```

Comment:
Please check the code section below. Pool1 and pool2 are deleted. However we are resetting full ratios on acting set of pool1. Pool3 is not yet deleted. Please fix them.

Reply:
Pool3 was added as additional check, removal of Pool2 and Pool3 is optional, but I will include it for uniformity

Reply:
Also full ratios are not pool specific, it is for the whole cluster
We are only reweighting the OSDs part of acting set of pool 1 because they are the only OSDs for which weights were modified. Line number 819-820 is redundant and only a fail safe mechanism to ensure that OSDs are not left weighted 0 in case test case fails in between for any reason.

Reply:
LGTM